Assets/Scripts/Augment/AugmentImplementations/DynamiteBarrel.cs

```diff
@@ -0,0 +1,137 @@
+    private bool isDetonatableEarly = false;
+    [SerializeField]
+    private MeshProjectileController meshProjectiles;
+    private SateliteUplink sateliteUplink;
+
+
+    void Start()
+    {
+        gunController = transform.parent.GetComponent<GunController>();
+        if (!gunController || !gunController.Player)
+            return;
+
+        gunController.Player.onDeath += OnDeath;
+        stickyModifer.OnStuckToTarget += AddDynamite;
+
+        // If bullets are large enough (enlarger), they should look like explosive barrels instead
+        if (Projectile.stats.ProjectileScale >= 3)
+            GetComponent<MeshProjectileController>().Vfx.SetMesh("Mesh", explodingBarrelMesh);
+
+        // If fire extension, dynamites should be able to explode while in air
+        if (gunController.GetComponentInChildren<Fire>())
+            isDetonatableEarly = true;
+        else
+            sateliteUplink = gunController.GetComponentInChildren<SateliteUplink>();
+
+        if (gunController.Player is AIManager)
+        {
+            StartCoroutine(TryDetonating());
+            return;
+        }
+        syncDirection = SyncDirection.ClientToServer;
+        gunController.Player.GetComponent<PlayerMovement>().ResetZoom();
+        gunController.Player.inputManager.onZoomPerformed += OnZoom;
+    }
+
+    private void AddDynamite(StuckObject stuckObject)
+    {
+        if (stuckObject is not StickyDynamite)
+            return;
+        var stuckDynamite = (StickyDynamite)stuckObject;
+        activeDynamites.Add(stuckDynamite);
+        if (Projectile.stats.ProjectileScale >= 3)
+            stuckDynamite.SetBarrel();
+        
+    }
+
+    // Letting the AIs also explode stuff every now and then
+    private IEnumerator TryDetonating()
+    {
+        while (true)
+        {
+            yield return new WaitForSeconds(2);
+            if (activeDynamites.Any())
+                activeDynamites.ForEach(dynamite => dynamite.Detonate(gunController.Player));
+            activeDynamites.Clear();
+        }
+    }
+
+    private void OnZoom(InputAction.CallbackContext ctx)
+    {
+        if (authority)
+            CmdDetonate();
+    }
+
+    [Command]
+    private void CmdDetonate()
+    {
+        RpcDetonate();
+    }
+
+    [ClientRpc]
+    private void RpcDetonate()
+    {
+        if (isDetonatableEarly)
+        {
+            var explosivePositions = meshProjectiles.ProjectilePositions;
+            explosivePositions.ToList().ForEach(position =>
+                {
+                    var dynamite = (StickyDynamite)stickyModifer.InstantiateManual(position);
+                    dynamite.Explosion.Init();
+                    activeDynamites.Add(dynamite);
+                });
+            meshProjectiles.ClearProjectiles();
+        }
+        else if (sateliteUplink)
+            sateliteUplink.TargetManual(activeDynamites
+                .Select(dynamite => dynamite.transform.position).ToArray());
```

Comment:
No need to pass an array, change type to IEnumerable and it accepts the linq query just fine

Reply:
Agree, only reason I didn't do this was because I was looking for a clean alternative to passing a collection into the method at all, but I didn't clean up after giving up lol